src/utils/api-request.ts

```diff
@@ -31,56 +31,79 @@ export type HttpMethod = 'GET' | 'POST' | 'PUT' | 'DELETE';
 /** API callback function type definition. */
 export type ApiCallbackFunction = (data: object) => void;
 
+/**
+ * Configuration for constructing a new HTTP request.
+ */
 export interface HttpRequestConfig {
   method: HttpMethod;
+  /** Target URL of the request. Should be a well-formed URL including protocol, hostname, port and path. */
   url: string;
   headers?: {[key: string]: string};
   data?: string | object | Buffer;
+  /** Connect and read timeout (in milliseconds) for the outgoing request. */
   timeout?: number;
 }
 
+/**
+ * Represents an HTTP response received from a remote server.
+ */
 export interface HttpResponse {
   readonly status: number;
   readonly headers: any;
+  /** Response data as a raw string. */
   readonly text: string;
+  /** Response data as a parsed JSON object. */
   readonly data: any;
 }
 
+interface LowLevelResponse {
+  status: number;
+  headers: http.IncomingHttpHeaders;
+  request: http.ClientRequest;
+  data: string;
+  config: HttpRequestConfig;
+}
+
+interface LowLevelError extends Error {
+  config: HttpRequestConfig;
+  code?: string;
+  request?: http.ClientRequest;
+  response?: LowLevelResponse;
+}
+
 class DefaultHttpResponse implements HttpResponse {
 
   public readonly status: number;
   public readonly headers: any;
   public readonly text: string;
 
-  private readonly data_: any;
-  private readonly request_: string;
+  private readonly parsedData: any;
+  private readonly parseError: Error;
+  private readonly request: string;
 
   constructor(resp: LowLevelResponse) {
```

Comment:
To be consistent, we have been adding javadocs in other files. Let's add one here.

Reply:
Done